examples/gopts.c

```diff
@@ -0,0 +1,241 @@
+ * @copyright BSD License (see LICENSE.md or https://www.libelektra.org)
+ */
+
+#include <kdb.h>
+#include <kdbease.h>
+#include <kdbhelper.h>
+
+#include <kdbopts.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+extern char ** environ;
+
+#define BASE_KEY "/sw/org/erm/#0/current"
+#define SPEC_BASE_KEY "spec" BASE_KEY
+
+static KeySet * createSpec (void)
+{
+	return ksNew (
+		10,
+		keyNew (SPEC_BASE_KEY "/emptydirs", KEY_META, "description", "remove empty directories", KEY_META, "opt", "d", KEY_META,
+			"opt/arg", "none", KEY_META, "opt/long", "dir", KEY_END),
+		keyNew (SPEC_BASE_KEY "/force", KEY_META, "description", "ignore nonexistent files and arguments, never prompt", KEY_META,
+			"opt", "f", KEY_META, "opt/arg", "none", KEY_META, "opt/long", "force", KEY_END),
+		keyNew (SPEC_BASE_KEY "/interactive", KEY_META, "description",
+			"prompt according to WHEN: never, once (-I), or always (-i), without WHEN, prompt always", KEY_META, "opt", "#1",
+			KEY_META, "opt/#0", "i", KEY_META, "opt/#0/arg", "optional", KEY_META, "opt/#0/flagvalue", "always", KEY_META,
+			"opt/#0/long", "interactive", KEY_META, "opt/#1", "I", KEY_META, "opt/#1/arg", "none", KEY_META, "opt/#1/flagvalue",
+			"once", KEY_META, "opt/arg/name", "WHEN", KEY_END),
+		keyNew (SPEC_BASE_KEY "/nopreserve", KEY_META, "description", "do not treat '/' specially", KEY_META, "opt/arg", "none",
+			KEY_META, "opt/long", "no-preserve-root", KEY_END),
+		keyNew (SPEC_BASE_KEY "/preserve", KEY_META, "description",
+			"do not remove '/' (default), with 'all', reject any command line argument on a separate device from its parent",
+			KEY_META, "opt/arg", "optional", KEY_META, "opt/arg/name", "all", KEY_META, "opt/flagvalue", "root", KEY_META,
+			"opt/long", "preserve-root", KEY_END),
+		keyNew (SPEC_BASE_KEY "/recursive", KEY_META, "description", "remove directories and their contents recursively", KEY_META,
+			"opt", "#1", KEY_META, "opt/#0", "r", KEY_META, "opt/#0/arg", "none", KEY_META, "opt/#0/long", "recursive",
+			KEY_META, "opt/#1", "R", KEY_META, "opt/#1/arg", "none", KEY_END),
+		keyNew (SPEC_BASE_KEY "/showversion", KEY_META, "description", "output version information and exit", KEY_META, "opt/arg",
+			"none", KEY_META, "opt/long", "version", KEY_END),
+		keyNew (SPEC_BASE_KEY "/singlefs", KEY_META, "description",
+			"when removing a hierarchy recursively, skip any directory that is on a file system different from that of the "
+			"corresponding line argument",
+			KEY_META, "opt/arg", "none", KEY_META, "opt/long", "one-file-system", KEY_END),
+		keyNew (SPEC_BASE_KEY "/verbose", KEY_META, "description", "explain what is being done", KEY_META, "opt", "v", KEY_META,
+			"opt/arg", "none", KEY_META, "opt/long", "verbose", KEY_META, "env", "VERBOSE", KEY_END),
+		keyNew (SPEC_BASE_KEY "/files/#", KEY_META, "description", "the files that shall be deleted", KEY_META, "args", "remaining",
+			KEY_META, "env", "FILES", KEY_END),
+		KS_END);
+}
+
+static int setupSpec (void)
```

Comment:
Please mention here that this is only like this to be self-contained, not that an application should do this.